src/gtkspellcheck/spellcheck.py

```diff
@@ -807,3 +807,71 @@ def _continue_batched_recheck(self, start_mark):
             end.forward_char()
             start_mark = self._buffer.create_mark(None, end)
             GLib.idle_add(self._continue_batched_recheck, start_mark)
+
+    @staticmethod
+    def _between_middle_and_end_of_word(loc):
+        if loc.starts_word():
+            if loc.is_start():
+                return False
+
+            # Determine if actually in word after extra chars
+            loc = loc.copy()
+            loc.backward_char()
+            if not SpellChecker._is_extra_word_char(loc):
+                return False
+
+            if loc.is_start():
+                return False
+            loc.backward_char()
+            return loc.inside_word()
+        else:
+            return loc.inside_word() or loc.ends_word()
+
+    @staticmethod
+    def _is_extra_word_char(loc):
+        ch = loc.get_char()
+
+        if ch == " " or ch == "\n" or ch == "\t" or ch == "\r":
+            return False
+
+        if ch == "'":
+            return True
+
+        # Language extra chararacters should also be processed once Enchant's
+        # enchant_dict_get_extra_word_characters is exposed in PyEnchant
+
+        return False
```

Comment:
Yeah, as I alluded to in the description this logic is a Python mirror of the C in Text Editor. I'm guessing maybe they wanted to avoid the presumably slower call to Enchant and make things very clear. One of the reasons I didn't minimise as you're talking about was to make it in theory possible in future to turn on the call to Enchant, but maybe that doesn't make sense.

I'll reduce as discussed.

Reply:
I see. Extending the function to use the characters provided by Enchant is still possible and a good idea. Please leave the comment there to indicate that this is where the future change should go.

> I'll reduce as discussed.

Thanks. :)

Reply:
Have reduced somewhat as requested. It ended up a little more complex to allow for the configured extra characters being in a string and checking the end of buffer (which returns an empty string).